### ISSUE TITLE
"Cartogram creation cancelled by user“?

### ISSUE THREAD

**Issue by cutterkom** [informeren/qgis-cartogram#4]
I am using the QGIS Cartogram plugin, but unfortunately something is not working as it should.

First, the demo works just fine. But when I want to create a cartogram with my own layer. But as soon as I click „ok“, it says "Cartogram creation cancelled by user“.

Do you have a clue, what I'm doing wrong? What am I canceling?

**Comment by Zaidai** [comment]
I experience the very same issue.

**Comment by cutterkom** [comment]
The reason: No NULL and zero values allowed.

**Comment by ebrelsford** [comment]
It would be great if these errors would be handled more gracefully as I suggest in #5. Better error messages would help too.

**Comment by wulff** [comment]
I totally agree. Currently the code fails with the same error message whether the action is cancelled by the user or if an error occurs. I'll split that error handling in to two separate parts.

What would be the best way to handle zero/NULL values? Should we just ignore features with zero/NULL values?

**Comment by ebrelsford** [comment]
I think ignoring zeros and NULLs and producing a gentle warning makes sense.

**Comment by wulff** [comment]
The latest code on the develop branch has improved handling of zero and NULL values. If they are present the user is given a choice to either abort cartogram creation or to modify the values so they don't break the algorithm.

**Comment by ebrelsford** [comment]
Great! Looking forward to trying it out.

**Comment by gurmanbh** [comment]
I don't have any null values. Still facing the same issue.

**Comment by prlitics** [comment]
gurmanbh, are you trying to create a cartogram using data from a joined file? That's the boat I was in-- I found I had to create a new shp file that had the data already within it.

**Comment by jorgeacast** [comment]
How did you create the new shp file?
